htdocs/mri/jiv/get_file.php

```diff
@@ -80,76 +73,65 @@
 }
 unset($path_parts);
 
-// Make sure that the user isn't trying to break out of the $path by
-// using a relative filename.
-// No need to check for '/' since all downloads are relative to $imagePath,
-// $DownloadPath or $mincPath
-if (strpos("..", $File) !== false) {
-    error_log("ERROR: Invalid filename");
-    header("HTTP/1.1 400 Bad Request");
-    exit(4);
-}
```

Comment:
What sort of problem with `realpath()` are you envisioning? Do you know of cases where it fails?

I don't like this bit of code because I think it provides a false sense of security. The comment is correct in that all downloads are relative to `$imagePath`, `$DownloadPath` or `$mincPath`. Therefore, the only piece we need to check is `$File`.

It feels more robust to resolve the path via `realpath()` rather than checking for literals in a string which is trivial to encode. When searching for ways to prevent path traversal, it was the combination of `realpath()` and `basename()` that came up as the preferred solution.